### PR TITLE
Show thankyou message and open downloads pages in new tabs

### PR DESCRIPTION
**Pull request by faebebin**
Improves #1079

### CODE REVIEW

**Comment by faebebin** [comment]
@agiudiceandrea all download links show thankyou and donate message now. The links that lead to a new page open in new tab. As the pages are internal I am not sure whether we still want this arrow icon (see screenshots  boelow). Should this indicate that it opens in new tab (keep it) or that linked resource is external (remove it)?

image


image

image

**Comment by agiudiceandrea** [comment]
In the alldownloads page at https://qgis.org/en/site/forusers/alldownloads.html I see also "internal" links with such icon, e.g. here:
image

Another approach would be to leave the OSGeo4W network installer link to open in the current tab (removing the external class) and make the "external" link in the alldownloads page to display the thankyou overlay when clicked.

**Comment by faebebin** [comment]
@agiudiceandrea I added the thankyou message to selected links in `/alldownloads` page including Windows, Linux and Mac resources. Does that make sense to you?

**Comment by agiudiceandrea** [comment]
Yes, it should be fine!
Anyway, in the meantime your previous PR https://github.com/qgis/QGIS-Website/pull/1078 seems to have fixed the decrease in donations reported by the qgis-donations bot in the previous month
image

**Comment by faebebin** [comment]
So @agiudiceandrea , @m-kuhn feel free to merge this?

File: themes/qgis-theme/languageswitch.html [hunk 1]
```diff
@@ -107,7 +107,22 @@
             '<br/><p><img src="{{pathto('_static/images/splash.png', 1)}}"/></p></div>');
             $('#thankyou').click(function(){ $('#thankyou').hide(); });
             $('#closethankyou').click(function(){ $('#thankyou').hide(); });
-            $('a.reference').click(function(){
+            var showTankyouElements = $('a.reference')
```
Comment (jef-n) [inline themes/qgis-theme/languageswitch.html]: typo

Comment (faebebin) [reply themes/qgis-theme/languageswitch.html]: ups, thanks

**Comment by m-kuhn** [comment]
I'd say let's merge and get donations back again

**Comment by faebebin** [comment]
> I'd say let's merge and get donations back again

@m-kuhn , as @agiudiceandrea mentioned above 
( https://github.com/qgis/QGIS-Website/pull/1082#issuecomment-1312476047 ) the fix was already effective. Merging this would display the message also for many linux links. So the question is if we want this enhancement or just keep it as was.

**Comment by m-kuhn** [comment]
Is there any downside to merging this?

**Comment by faebebin** [comment]
It adds (little) more code and complexity to Javascript. So I would only merge it if these additional `Thankyou`-messages on the alldownloads-page really add value.

File: themes/qgis-theme/forusers_download.html [hunk 2]
```diff
@@ -46,7 +46,7 @@ <h4>{{ _('Download for Windows') }} <i class="icon-chevron-down pull-right"></i>
                         {{ _('Looking for the most stable version? Get QGIS %s %s')|format( ltrversion, ltrnote ) }}
                       </a>
                       <br/>
-                      <a class="reference secondary-download-link" href="{{ pathto('site/forusers/alldownloads') }}#osgeo4w-installer">
+                      <a class="secondary-download-link" href="{{ pathto('site/forusers/alldownloads') }}#osgeo4w-installer">
                         <div class="primary-download-button">
```
Comment (faebebin) [inline themes/qgis-theme/forusers_download.html:50]: This is a necessary fix.

File: themes/qgis-theme/languageswitch.html [hunk 3]
```diff
@@ -107,7 +107,22 @@
...
+                'section#fedora a.reference, '+
+                'section#suse-opensuse a.reference, '+
+                'section#mandriva a.reference, '+
+                'section#slackware a.reference, '+
+                'section#arch-linux a.reference, '+
+                'section#flatpack a.reference:contains("qgis"), '+
+                'section#mac-os-x-macos a.reference:contains("kyngchaos"), '+
+                'section#mac-os-x-macos #qgis-nightly-release a.reference '
+              )
+            }
```
Comment (faebebin) [inline themes/qgis-theme/languageswitch.html]: This whole selection could be discussed. Could also be simplified to only show the Thankyou message for the `section#windows` links (ie. osgeo4w installer).

Comment (DelazJ) [reply themes/qgis-theme/languageswitch.html]: These links, If I'm correct, just lead to a new page (in our website) with kind of instructions to get QGIS. OTOH The message that pops up congratulates the user for being downloading the binaries (which is quite wrong). Not the same thing. IMHO it is confusing to display the thank you message for Linux links

Comment (faebebin) [reply themes/qgis-theme/languageswitch.html]: Yes agree. Will change it to only show on the windows links that actually start the package download.

Comment (DelazJ) [reply themes/qgis-theme/languageswitch.html]: > to only show on the windows links

windows and macos (in the main page), right?!

Comment (agiudiceandrea) [reply themes/qgis-theme/languageswitch.html]: > windows and macos (in the main page), right?!

The Windows and macOS binaries links in the main download page already display the Thankyou/Donate message (after https://github.com/qgis/QGIS-Website/pull/1078).

Comment (faebebin) [reply themes/qgis-theme/languageswitch.html]: Sorry for confusion. Will update the mentioned correction asap to clearify.

Comment (faebebin) [reply themes/qgis-theme/languageswitch.html]: Sorry for confusion. Will update the mentioned correction asap to clearify.

**Comment by faebebin** [comment]
No it only shows Thankyou message only for the links that actually start the installer download, which are 3 links on the `/alldownloads.html` windows section.

Screenshot from 2022-11-16 14-39-14

File: themes/qgis-theme/languageswitch.html (hunk 3 above)
Comment (faebebin) [reply themes/qgis-theme/languageswitch.html]: It now only shows Thankyou message for all links that actually start an installer download, which are 3 links on the `/alldownloads.html` windows section and the windows and mac downloads on the main `/download.html` page

Screenshot from 2022-11-16 14-39-14

Comment (DelazJ) [reply themes/qgis-theme/languageswitch.html]: > It now only shows Thankyou message for all links that actually start an installer download

Fine with me. Thank you @faebebin

**Review by agiudiceandrea — APPROVED** [review]
It seems OK to me.

**Comment by m-kuhn** [comment]
Thanks for the work and all the reviews!

**Pull request merged**